### PR TITLE
Mkdocs

### PR DESCRIPTION
**Pull request by signedav**
Moving updated ModelBaker Documentation to this place.

The "old" repository (https://github.com/opengisch/QgisModelBaker-docs) won't be used after.

## Setup Gh-Pages
- Need to create gh-pages branch run on /root

## Test it locally
- To test locally do `mkdocs serve`

## Transifex Project
- The Transifex Project "Qgis Model Baker Docs" is the one these files are pushed to.
- The "old" Project "QGis Model Baker Documentation" won't be used after.

## To Does
- [x] Translate Usabilityhub Docus to English
- [x] UsabILIty Hub (check 6.5.0 changelog, PR #468 and usabilityhub.opengis.ch documents and clickup notes)
- [x] Usabilityhub cat handling
- [x] Usability Hub is still german
- [x] General Translation stuff
- [x] Basket Handling: Import/Export with Baskets (changelog 6.6)
- [x] Basket Handling: Dataset Manager (changelog 6.6) 
- [x] Basket Handling: Dataset Selector  (changelog 6.6)
- [x] Wizard

### CODE REVIEW

File: docs/background_info/meta_attributes.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+> To a metaelement metaattributes could be compositional assigned (class "MetaAttribut"). The metaattributes are neither defined by the language INTERLIS nor by the meta model. They exist that the information that exceed INTERLIS could still be a component of the model data. The metaattributes has a name (attribute "Name"), that have to be unique among the the metaattributes of this metaelement, and it has a value (attribute: "Value").
```
Comment (sjib) [inline docs/background_info/meta_attributes.md:1]: "A meta element can be assigned compositional meta attributes (class "MetaAttribute"). Metaattributes are not defined more precisely by the INTERLIS language or by the metamodel. Their main purpose is to allow information that goes beyond INTERLIS to be attached as part of the model data. Metaattributes have a name (attribute "Name"), which must be unique within the metaattributes of the same meta element, and the descriptive value (attribute "Value")."

INTERLIS is always in capital letters

File: docs/background_info/meta_attributes.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+Translated from source: [STAN_d_DEF_2011-06-22_eCH-0117 Meta-Attribute für INTERLIS-Modelle.pdf](https://www.ech.ch/alfresco/s/ech/download?nodeid=788eb38a-bf2b-4f3d-96a8-addc37bba41f) (German) from eCH www.interlis.ch
```
Comment (sjib) [inline docs/background_info/meta_attributes.md:3]: The original source of this quote is in [2] KOGIS (2008): INTERLIS 2-Metamodell.
Online http://www.interlis.ch/interlis2/docs23/ili2-metamodel_2008-08-
29_d.pdf

Im Dokument zum INTERLIS 2-Metamodell [2] steht dazu:
«Einem Metaelement können kompositorisch Metaattribute zugeordnet sein (Klasse "MetaAttribut").
Metaattribute sind weder durch die Sprache INTERLIS noch durch das Metamodell
genauer definiert. Sie dienen insbesondere dazu, dass Informationen, die über INTERLIS
hinausgehen, als Bestandteil der Modelldaten beigefügt werden können. Metaattribute
haben einen Namen (Attribut "Name"), der innerhalb der Metaattribute desselben Metaelementes
eindeutig sein muss, und den beschreibenden Wert (Attribut "Value").»

File: docs/background_info/meta_attributes.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    EXTENDS CatalogueObjects_V1.Catalogues =
+    !!@ dispExpression="CONCAT(type, ' sometext')"
+    CLASS TypeOfRoute
+    EXTENDS CatalogueObjects_V1.Catalogues.Item =
+```
+
+`furtherInformation` is referenced to `ExceptionalLoadsCatalogues_V1`, `topicInformation` to `TypeOfRouteCatalogue` and `dispExpression` to `TypeOfRoute`.
+
+For more complex usage see the [specification of the Verein eCH](https://www.ech.ch/alfresco/s/ech/download?nodeid=788eb38a-bf2b-4f3d-96a8-addc37bba41f).
+
```
Comment (sjib) [inline docs/background_info/meta_attributes.md:40]: The link does not work - when I look it up on https://ech.ch/de/standards/60625 I find this link:
https://ech.ch/index.php/de/dokument/1f67b229-f6bc-4fb4-a437-f91602e0b01d

File: docs/background_info/meta_attributes.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
+> To a metaelement metaattributes could be compositional assigned (class "MetaAttribut"). The metaattributes are neither defined by the language INTERLIS nor by the meta model. They exist that the information that exceed INTERLIS could still be a component of the model data. The metaattributes has a name (attribute "Name"), that have to be unique among the the metaattributes of this metaelement, and it has a value (attribute: "Value").
+
+Translated from source: [STAN_d_DEF_2011-06-22_eCH-0117 Meta-Attribute für INTERLIS-Modelle.pdf](https://www.ech.ch/alfresco/s/ech/download?nodeid=788eb38a-bf2b-4f3d-96a8-addc37bba41f) (German) from eCH www.interlis.ch
+
+## Meta Attributes in Interlis Files
```
Comment (sjib) [inline docs/background_info/meta_attributes.md:5]: INTERLIS is always in capital letters

File: docs/background_info/meta_attributes.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+### Comment vs. Meta Attribute
+
+An Interlis comment starts with `!!` and ends with a line end. A meta attribute starts with `!!` as well but followed by an `@`:
```
Comment (sjib) [inline docs/background_info/meta_attributes.md:9]: INTERLIS is always in capital letters

File: docs/background_info/meta_attributes.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    EXTENDS CatalogueObjects_V1.Catalogues.Item =
+```
+
+`furtherInformation` is referenced to `ExceptionalLoadsCatalogues_V1`, `topicInformation` to `TypeOfRouteCatalogue` and `dispExpression` to `TypeOfRoute`.
+
+For more complex usage see the [specification of the Verein eCH](https://www.ech.ch/alfresco/s/ech/download?nodeid=788eb38a-bf2b-4f3d-96a8-addc37bba41f).
+
+### Meta Attributes in the Database
+
+When importing the data from the Interlis file to the DB, with ili2db integrated in the Model Baker, the meta attributes are stored in the table **t_ili2db_meta_attrs**:
```
Comment (sjib) [inline docs/background_info/meta_attributes.md:43]: INTERLIS is always in capital letters

File: docs/background_info/meta_attributes.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+### List of specific Attributes
+
+- **dispExpression**
+Used as the display expression for a layer. The display expression is the *name* that is used to identify a feature by text. One of the places where this is used is the combobox that is shown for a Relation Reference Widget on feature forms. ![relation reference](../assets/meta_attributes_relation_reference.png)
+
+
+## Extra Model Information File
+
+In these external files the meta attributes can be stored instead of having them directly in the Interlis files.
```
Comment (sjib) [inline docs/background_info/meta_attributes.md:63]: INTERLIS is always in capital letters

File: docs/background_info/meta_attributes.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+["ExceptionalLoadsRoute.TypeOfRouteCatalogue.TypeOfRoute"]
+dispExpression="type"
+```
+Or using a more complex expression:
+```ini
+["ExceptionalLoadsCatalogues_V1.TypeOfRouteCatalogue.TypeOfRoute"]
+dispExpression="CONCAT(type, ' sometext')"
+```
+
+The keys that need to be used for the TOML sections are the *fully qualified Interlis names* of the objects. In the example above this is `["ExceptionalLoadsCatalogues_V1.TypeOfRouteCatalogue.TypeOfRoute"]`. A list of all available names can be found in the database table `t_ili2db_classname` after doing a schema import.
```
Comment (sjib) [inline docs/background_info/meta_attributes.md:86]: INTERLIS is always in capital letters

File: docs/background_info/meta_attributes.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+### Interlis Example
```
Comment (sjib) [inline docs/background_info/meta_attributes.md:88]: INTERLIS is always in capital letters

File: docs/background_info/meta_attributes.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+
+The above example would be written as follows directly in an Interlis file (ExceptionalLoadsRoute.ili). If the above configuration is in the ini file, the ili meta attribute is no longer required, this is only here for reference.
```
Comment (sjib) [inline docs/background_info/meta_attributes.md:90]: INTERLIS is always in capital letters

File: docs/user_guide/get_started.md [hunk 11]
```diff
@@ -0,0 +1,13 @@
+## Installation
+
+1. [Install QGIS 3](https://qgis.org/en/site/forusers/download.html)
+
+2. Use the plugin manager to install the "QGIS Model Baker" plugin
+
+3. Click the *Database > QGIS Model Baker > Generate* menu
+
+4. Configure your database and, if applicable, Interlis configuration
```
Comment (sjib) [inline docs/user_guide/get_started.md:9]: INTERLIS is always in capital letters

File: docs/user_guide/plugin_configuration.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
+Go to menu *Database > Model Baker > Settings*.
+
+## General Settings
+![general settings](../assets/settings_general.png)
+
+In keeping with aristotle, unnecessary settings are avoided. Still there are some.
+
+In the *Database Login* you can set username and password to use it as superuser login to excecute data management tasks. The *checkbox* is to avoid / enforce opening a popup when dropping XTF/ITF or ILI files in QGIS main window to ask if you want to handle them in the Model Baker or not.
+
+## Interlis Settings
```
Comment (sjib) [inline docs/user_guide/plugin_configuration.md:10]: INTERLIS is always in capital letters

File: docs/user_guide/plugin_configuration.md [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+
+![interlis settings](../assets/settings_interlis.png)
```
Comment (sjib) [inline docs/user_guide/plugin_configuration.md:12]: INTERLIS is always in capital letters

File: docs/user_guide/plugin_configuration.md [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+### Custom Model Directories
+
+In this window you can set one or several paths to let ili2db where to find your Interlis models (ili files). Paths can be both local directories or URLs to remote model repositories. Several paths should be separated by semicolons.
```
Comment (sjib) [inline docs/user_guide/plugin_configuration.md:15]: INTERLIS is always in capital letters

File: docs/background_info/meta_attributes.md (hunk 1 above)
Comment (signedav) [reply docs/background_info/meta_attributes.md]: Thanks. Considered that and all the INTERLIS uppercase issues.

**Comment by signedav** [comment]
@marioba does the setup looks good?

File: .github/workflows/doc.yml [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.x
+      - run: pip install mkdocs-material
```
Comment (marioba) [inline .github/workflows/doc.yml:17]: Isn't the "i18n" plugin missing?

**Review by marioba — COMMENTED** [review]
Wouldn't it be better to have a requirements.txt (or somewhere else) with the dependencies needed to generate the documentation?

File: .github/workflows/doc.yml (hunk 15 above)
Comment (signedav) [reply .github/workflows/doc.yml:17]: Yes. Think I mixed up something in the final commit here https://github.com/opengisch/QgisModelBaker/pull/625/commits/1dbdbf5c09dc4ef778e82d9aa58992fcd322eda0 Remember that I tried something out with requirements etc.

**Comment by signedav** [comment]
Does the Transifex Setup make sence to you @m-kuhn ?

**Comment by signedav** [comment]
@m-kuhn and can you give me rights to publish the gh-pages (enter the settings)

File: .github/workflows/doc.yml [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+        with:
+          python-version: 3.x
+
+      - name: Install Requirements
+        run: |
+            sudo apt-get update && sudo apt-get install -y transifex-client
+            sudo pip install mkdocs-material mkdocs-static-i18n==0.31
+
+      - name: Push translations
+        if: ${{ github.ref == 'refs/heads/master' }}
```
Comment (m-kuhn) [inline .github/workflows/doc.yml:27]: Are we sure that doesn't run for pull requests?

File: .tx/config [hunk 17]
```diff
@@ -0,0 +1,109 @@
+[main]
+host = https://www.transifex.com
+
+[qgis-model-baker-docs.index-md]
```
Comment (m-kuhn) [inline .tx/config:4]: @marioba is there a possibility to generalize this? it seems this file is very likely to be forgotten when new chapters are added to the doc

File: mkdocs.yml [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+  - Tipps & Tricks:
+    - Repositories: background_info/repositories.md
+    - Basket and Dataset Handling: background_info/basket_handling.md
+    - UsabILIty Hub:
+      - Model Baker Integration (German): background_info/usabilityhub/modelbaker_integration.md
+      - Technical Concept (German): background_info/usabilityhub/technical_concept.md
+      - User Guide (German): background_info/usabilityhub/user_guide.md
+    - Meta Attributes: background_info/meta_attributes.md
+    - Upgrade ili2db 3 to 4: background_info/upgrade_3_to_4.md
+    #- Relations in QGIS: maybe stuff from here https://github.com/signedav/interlis_relations_in_qgis
```
Comment (m-kuhn) [inline mkdocs.yml:48]:
```suggestion
    #- Relations in QGIS: maybe from here https://github.com/signedav/interlis_relations_in_qgis
```

File: mkdocs.yml [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+    - Plugin Configuration: user_guide/plugin_configuration.md
+  - Tipps & Tricks:
+    - Repositories: background_info/repositories.md
+    - Basket and Dataset Handling: background_info/basket_handling.md
+    - UsabILIty Hub:
+      - Model Baker Integration (German): background_info/usabilityhub/modelbaker_integration.md
+      - Technical Concept (German): background_info/usabilityhub/technical_concept.md
+      - User Guide (German): background_info/usabilityhub/user_guide.md
+    - Meta Attributes: background_info/meta_attributes.md
+    - Upgrade ili2db 3 to 4: background_info/upgrade_3_to_4.md
```
Comment (m-kuhn) [inline mkdocs.yml:47]:
```suggestion
    - Migrate from ili2db 3 to 4: background_info/upgrade_3_to_4.md
```

File: .tx/config (hunk 17 above)
Comment (signedav) [reply .tx/config:4]: Yes we can. I will add the command:
```
tx config mapping-bulk -p qgismodelbaker-docs --source-language en --type MD -f '.md' --source-file-dir docs --expression "docs/{filepath}/{filename}.<lang>{extension}" --execute
```
to the workflow

File: .github/workflows/doc.yml (hunk 16 above)
Comment (signedav) [reply .github/workflows/doc.yml:27]: I think so. No?

File: mkdocs.yml [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+    - Basket and Dataset Handling: background_info/basket_handling.md
+    - UsabILIty Hub:
+      - Model Baker Integration: background_info/usabilityhub/modelbaker_integration.md
+      - Technical Concept: background_info/usabilityhub/technical_concept.md
+      - User Guide: background_info/usabilityhub/user_guide.md
+    - Meta Attributes: background_info/meta_attributes.md
+    - Migrate from ili2db 3 to 4: background_info/upgrade_3_to_4.md
+    #- Relations in QGIS: maybe from here https://github.com/signedav/interlis_relations_in_qgis
+    #- INTERLIS Syntax in 10 Minutes: maybe stuff from here https://github.com/signedav/talk_iliuniverse
+ #- Library: Info about the upcoming python package
```
Comment (m-kuhn) [inline mkdocs.yml:50]: Should go under the developer section

**Review by m-kuhn — APPROVED** [review]

**Comment by signedav** [comment]
Can you activate the gh-pages for that on /root @m-kuhn (or give me rights)?

**Comment by signedav** [comment]
And I need to set a TX_TOKEN - I could set mine, but I expected that there is already one (but possibly was removed when plugin-ci was introduced).

**Pull request merged**